### DISCUSSION TITLE
Can the unused imports recepie remove wildcard includes?

### DISCUSSION THREAD

**Discussion by agreedSkiing** [openrewrite/rewrite#2728] [Q&A] [answered]
Can the recepie that removes unused imports remove imports like `import java.lang.*`

**Comment by timtebeek** [comment, marked as answer]
Hi @agreedSkiing ; you might want to have a look at [Order Imports](https://docs.openrewrite.org/reference/recipes/java/orderimports) instead; that one also has to option to remove unused imports, and should expand star imports.

There could be a bit more nuance to it, as we typically try to [AutoDetect](https://github.com/openrewrite/rewrite/blob/v7.35.0/rewrite-java/src/main/java/org/openrewrite/java/style/Autodetect.java) the [Style](https://docs.openrewrite.org/concepts-and-explanations/styles) used in a project, and conform to the prevailing style found throughout the project already. If you want more control over expanding imports, you can look into defining an explicit style; we support a couple common ones already, but you can define custom styles as well.
